lib/storage/metadata/mongoclient/MongoClientInterface.js

```diff
@@ -751,20 +895,30 @@ class MongoClientInterface {
                 });
     }
 
-    /*
+    /**
      * repair the master with a new value. There can be
      * race-conditions or legit updates so place an atomic condition
      * on PHD flag and mst version.
+     * @param {Object} c collection
+     * @param {String} bucketName bucket name
+     * @param {String} objName object name
+     * @param {Object} objVal new object version metadata
+     * @param {Object} mst master version metadata
+     * @param {String} vFormat key format version
+     * @param {Object} log logger
+     * @param {Function} cb callback
+     * @return {undefined}
      */
-    repair(c, bucketName, objName, objVal, mst, log, cb) {
+    repair(c, bucketName, objName, objVal, mst, vFormat, log, cb) {
```

Comment:
In metadata implementation, v1 does not create any PHD key since the repair is done at time of deletion of a specific version, hence the repair code should not have to deal with v1. But if you decided to keep the PHD in v1 in the Mongo implementation, then you can ignore my comment.

Reply:
We could get rid of the PHDs in MongoClientInterface but I suggest we do that in a v2:
E.g.
- Use transactions to atomically update/delete MasterKey + LastVersionKey MD (the rationale is to have the MD up-to-date at all time in the MasterKey for O(1) access of the MD.

So let's make it simple and fix the DeleteMarkers first in v1 (and therefore keep PHD related concepts).